### PR TITLE
add refferal end at to prevent expired intereventions from being returned

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Stops interventions that have ended being returned in results

## What is the intent behind these changes?

Allows interventions to be given a specific time to no longer be seen

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImpl.kt [hunk 1]
```diff
@@ -103,4 +105,11 @@ class InterventionFilterRepositoryImpl(
       criteriaBuilder.lessThanOrEqualTo(expression, startDate)
     }
   }
+
+  private fun removeEndedContracts(criteriaBuilder: CriteriaBuilder, root: Root<Intervention>): Predicate? {
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImpl.kt:109]: nice name

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImplTest.kt [hunk 2]
```diff
@@ -237,4 +238,18 @@ class InterventionFilterRepositoryImplTest @Autowired constructor(
     assertThat(found.size).isEqualTo(4)
     assertThat(found).contains(futureIntervention)
   }
+
+  @Test
+  fun `get interventions with an ended contract in the past and one in the future`() {
```
Comment (sldblog) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImplTest.kt:243]: I like this assertion at the end.
```
    assertThat(found).doesNotContain(endedIntervention)
    assertThat(found).contains(futureEndIntervention)
```

Could this be reflected in the test name, with assertive language (e.g. "filters out", "removes"), please?

**Review by sldblog — COMMENTED** [review]
LGTM!
Just one comment about test case naming (for more straightforward understanding)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImplTest.kt (hunk 2 above)
Comment (rgforsyth) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImplTest.kt]: updated.

**Review by sldblog — APPROVED** [review]
👍 LGTM, nice

**Pull request merged**